synthtool/languages/node_mono_repo.py

```diff
@@ -349,12 +351,32 @@ def walk_through_owlbot_dirs(dir: Path):
     """
     owlbot_dirs = []
     packages_to_exclude = [r"gapic-node-templating", r"node_modules"]
+    if search_for_changed_files:
+        subprocess.run(["git", "fetch", "origin", "main", "--deepen=200"])
     for path_object in dir.glob("packages/**/.OwlBot.yaml"):
         if path_object.is_file() and not re.search(
             "(?:% s)" % "|".join(packages_to_exclude), str(Path(path_object))
         ):
-            owlbot_dirs.append(str(Path(path_object).parents[0]))
-
+            if search_for_changed_files:
+                if (
+                    subprocess.run(
+                        [
+                            "git",
+                            "diff",
+                            "--quiet",
+                            "origin/main...",
+                            Path(path_object).parents[0],
+                        ]
+                    ).returncode
+                    == 1
+                ):
+                    owlbot_dirs.append(str(Path(path_object).parents[0]))
+            else:
+                owlbot_dirs.append(str(Path(path_object).parents[0]))
+    for path_object in dir.glob("owl-bot-staging/*"):
```

Comment:
I like this approach, you don't even need to use the `git` call, you just need to check whether `owl-bot-staging` exists 😄